### ISSUE TITLE
🐛 [firebase_auth] Version 4.5.0 depends on firebase_auth_web ^5.4.0, which depends on intl ^0.17.0

### ISSUE THREAD

**Issue by carloshpb** [firebase/flutterfire#10938]
## Bug report

Resolving dependencies...
Because no versions of firebase_auth match >4.5.0 <5.0.0 and firebase_auth 4.5.0 depends on firebase_auth_web ^5.4.0, firebase_auth ^4.5.0 requires firebase_auth_web ^5.4.0.
Because firebase_auth_web 5.4.0 depends on intl ^0.17.0 and no versions of firebase_auth_web match >5.4.0 <6.0.0, firebase_auth_web ^5.4.0 requires intl ^0.17.0.
Thus, firebase_auth ^4.5.0 requires intl ^0.17.0.
So, because __ depends on both firebase_auth ^4.5.0 and intl ^0.18.0, version solving failed.

### Steps to reproduce

1. Simply try to updated firebase_auth while having intl version to 0.18, which is the one for Dart 3

### Expected behavior

Shouldn't have conflict with intl version 0.18

**Comment by Patrick386** [comment]
Same issue.
Because no versions of firebase_auth match >4.5.0 <5.0.0 and firebase_auth 4.5.0 depends on firebase_auth_web ^5.4.0, firebase_auth ^4.5.0 requires firebase_auth_web ^5.4.0.
Because firebase_auth_web 5.4.0 depends on intl ^0.17.0 and no versions of firebase_auth_web match >5.4.0 <6.0.0, firebase_auth_web ^5.4.0 requires intl ^0.17.0.
Thus, firebase_auth ^4.5.0 requires intl ^0.17.0.
So, because __ depends on both intl ^0.18.0 and firebase_auth ^4.5.0, version solving failed.

**Comment by mdddj** [comment]
```yaml
dependency_overrides:
  intl: 0.18.0
```
Solved my problem

**Comment by BirjuVachhani** [comment]
This doesn't solve the problem for me unfortunately. I use wiredash and it creates conflicts with this for some reason.

**Comment by immadisairaj** [comment]
Maybe to add some more context on `firebase_ui_auth` as well that I am currently facing after upgrading all my `firebase` packages.

```sh
Because no versions of firebase_ui_auth match >1.3.0 <2.0.0 and firebase_ui_auth 1.3.0 depends on flutter_localizations from sdk, firebase_ui_auth ^1.3.0 requires flutter_localizations from sdk.
And because every version of flutter_localizations from sdk depends on intl 0.18.0, firebase_ui_auth ^1.3.0 requires intl 0.18.0.
And because firebase_auth_web 5.4.0 depends on intl ^0.17.0 and no versions of firebase_auth_web match >5.4.0 <6.0.0, firebase_ui_auth ^1.3.0 is incompatible with firebase_auth_web ^5.4.0.
And because firebase_auth 4.5.0 depends on firebase_auth_web ^5.4.0 and no versions of firebase_auth match >4.5.0 <5.0.0, firebase_ui_auth ^1.3.0 is incompatible with firebase_auth ^4.5.0.
```

<details><summary>Packages (and their versions) I am using</summary>
<p>

```yaml
cloud_firestore: ^4.6.0
firebase_auth: ^4.5.0
firebase_core: ^2.11.0
firebase_ui_auth: ^1.3.0
firebase_ui_firestore: ^1.4.0
firebase_ui_oauth_google: ^1.1.5
```

</p>
</details>

**Comment by rayk** [comment]
> ```yaml
> dependency_overrides:
>   intl: 0.18.0
> ```
> 
> Solved my problem

This will work if you don't depend on flutter_localizations and SDK is at 3.0.0. So the problem for any working off stable with an intl app.

**Comment by immadisairaj** [comment]
Yeah, that solved.. and the thing regarding `flutter_localizations` I solved it by adding it to the pubspec.

```yaml
dependencies:
  # for localizations (firebase_ui_auth uses)
  flutter_localizations:
      sdk: flutter

# for rest
dependency_overrides:
  into: 0.18.0
```

**Comment by krunalSutariya** [comment]
dependency_overrides:
  intl: 0.18.0
Solved my problem

**Comment by BirjuVachhani** [comment]
> Yeah, that solved.. and the thing regarding `flutter_localizations` I solved it by adding it to the pubspec.
> 
> ```yaml
> dependencies:
>   # for localizations (firebase_ui_auth uses)
>   flutter_localizations:
>       sdk: flutter
> 
> # for rest
> dependency_overrides:
>   intl: 0.18.0
> ```

This worked for me. However, my project uses melos and one of the deployed sub-packages uses older version of intl (0.17.0) and for some reason, melos doesn't seem to work with it. I had to manually create `pubspec_overrides.yaml`.

**Comment by carloshpb** [comment]
Nevermind guys. New updated was just launched right now. Just update your firebase_auth to 4.6.0, and let intl be 1.8.0, that it'll work out.
